EIPS/eip-2225.md

```diff
@@ -0,0 +1,126 @@
+
+We define methods to merge multiple blockchains together, in a fully
+backward compatible way.
+
+## Abstract
+
+This defines how Ethereum merging can happen, with backward
+compatibility in mind. This is done by utilizing account versioning
+together with transaction replay protection. We also define three
+additional opcodes to allow merged chains to work across two sides of
+state.
+
+## Motivation
+
+There are many reasons why we may want to merge multiple Ethereum
+(eth1.x) blockchains together. For example, there may have been chain
+splits of a blockchain, where later the community issues are resolved,
+and we want to avoid a permanent split due to a temporary issue. It
+may also be two completely distinct blockchains that want to merge in
+order to share proof of work, or to improve inter-chain communication.
+
+## Specification
+
+The merge process happens at fork block, where we have a **merger**
+chain and a **mergee** chain. At fork height, a miner should mine both
+a merger chain's block, and a mergee chain's block. For those two
+blocks:
+
+* Both of them should have no transactions.
+* Merger chain's fork height block and mergee chain's fork height
+  block should have its state root processed as normal.
+* Merger chain's fork height block should set its extra data field to
+  mergee chain's fork height block's state root. The same for mergee
+  chain's fork height block.
+* After mergee chain's fork height block, its proof of work should be
+  abandoned. That is, updated miners should not continue to mine
+  mergee chain any more.
+  
+### State Root
+  
+For the block after merger chain's fork height block, the state root
+meaning is changed to be hash of concatenating merger chain's original
+state root, and mergee chain's original state root.
+
+### Block Rewards
+
+Use merger chain's original block rewards. And mine coinbase always to
+the merger chain's side of the state.
+
+### Transactions and Account Versioning
+
+After merge, transactions with both chain ids are considered valid on
+the merger chain. We define two account versions, one with merger
+chain's original EVM config, and one with mergee chain's original EVM
+config. Accessing accounts originally from either chain should use
+EIP-1702's account versioning rules. We define how accounts are
+referred on the merged blockchain below:
+
+* For end user interface, prefix `0x01` for accounts on the merger
+  chain, and prefix `0x02` for accounts on the mergee chain.
+* The transaction format is unchanged with addresses using original
+  format. We distinguish whether a transaction operates on merger chain
+  or mergee chain by its chain id. None replay protected transactions
+  must be disabled prior to merging.
+* All original opcodes `CALL*`, `CREATE*` are unchanged, where it only
+  accesses accounts on the same originating chain.
+  
+### Cross the Merger and Mergee Chain Boundary
+
+Define factor `P // Q`, which determines how coins are exchanged on
```

Comment:
what are the variables `P` and `Q`?

Reply:
> Ok so this also proposes a massive increase in the total inflation?

I think that is subject to negotiation (something that would be very chaotic in decentralized systems, but very interesting as a thought experiment).

I see two models:

1. Where both currencies are preserved and the new merged chain continues as a bimonetary system. This is not "massive inflation" it's just adding two currencies. However one may be redundant just because before there were 2 underlying systems and now only one. 

2. Where the currencies are merged into one new currency using their pre-merger market caps as proxy for distribution of the new coins. In this, two coins that have ~110 million in stock each can be merged into a single stock of, say, 100 million where holders of each coin get proportional coins depending on their pre-merger market cap proportions.

Mathematically, both cases are identical, that is, nobody loses value, but the second case eliminated bimonetarism and creates a single new currency.

On the other hand, the above currency merger models have the problem that one or the other currency may be at a disadvantage if the market caps are very different, for example ETH is 32-to-1 with respect to ETC. In that case it's a matter of negotiation and for both ecosystems to set an exchange rate that is convenient for both.

Reply:
It all depends on whether the merger is accretive. If both systems together are worth more than the sum of their part before the merger, then it would be convenient for both ecosystems.

OTOH, there are roadmaps, philosophies and future monetary policies to homogenize in case of merger.

Reply:
hmmm thats interesting...

#1 is interesting, im assuming one of them will need to be the reward token